vars/buildPlugin.groovy

```diff
@@ -121,7 +123,28 @@ def call(Map params = [:]) {
                   }
                   mavenOptions += 'clean install'
                   try {
-                    infra.runMaven(mavenOptions, jdk, null, null, addToolEnv)
+                    if (artifactCachingProxyEnabled) {
+                      // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
+                      final String defaultProxyProvider = 'azure'
+                      def availableProxyProviders = ['azure', 'do', 'aws']
+                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
+                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
+                        requestedProvider = defaultProxyProvider
+                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      } else {
+                        echo "INFO: using artifact caching proxy from '${requestedProvider}' provider."
+                      }
+
+                      configFileProvider(
+                          [configFile(fileId: "artifact-caching-proxy-${requestedProvider}", variable: 'MAVEN_SETTINGS')]) {
+                            infra.runMaven(mavenOptions, jdk, null, env.MAVEN_SETTINGS, addToolEnv)
+                          }
+                    } else {
+                      echo 'NOTICE: artifacts will be downloaded directly from https://repo.jenkins-ci.org.'
```

Comment:
probably don't need to spam builds with this? but doesn't matter much

Reply:
You're right it doesn't make sense to add this line while the functionality is opt-in, neither to nag people who would have opted out when it's switched to opt-out.